python/route_guide/README.md

```diff
@@ -0,0 +1,281 @@
+   - `EarlyAdopterRouteGuideServer` which may be started and stopped
+   - `EarlyAdopterRouteGuideStub` which may be used to invoke RouteGuide RPCs
+- functions for application use
+   - `early_adopter_create_RouteGuide_server` which creates a server given a servicer object
+   - `early_adopter_create_RouteGuide_stub` which creates a stub object
+
+<a name="server"></a>
+## Creating the server
+
+First let's look at how you create a `RouteGuide` server. If you're only interested in creating gRPC clients, you can skip this section and go straight to [Creating the client](#client) (though you might find it interesting anyway!).
+
+Running a `RouteGuide` server breaks down into two work items:
+- Implementing the servicer interface generated from our service definition with functions that perform the actual "work" of the service.
+- Running a gRPC server to listen for requests from clients and transmit responses.
+
+You can find the example `RouteGuide` server in [grpc-common/python/route_guide/route_guide_server.py](https://github.com/grpc/grpc-common/blob/master/python/route_guide/route_guide_server.py).
+
+### Implementing RouteGuide
+
+`route_guide_server.py` has an `RouteGuideServicer` class that implements the generated interface `route_guide_pb2.EarlyAdopterRouteGuideServicer`:
+
+```python
+# RouteGuideServicer provides an implementation of the methods of the RouteGuide service.
+class RouteGuideServicer(route_guide_pb2.EarlyAdopterRouteGuideServicer):
+```
+
+`RouteGuideServicer` implements all the service methods. Look at the simplest type first, `GetFeature`, which just gets a `Point` from the client and returns the corresponding feature information from its database in a `Feature`.
+
+```python
+  def GetFeature(self, request, context):
+    feature = get_feature(self.db, request)
+    if feature is None:
+      return route_guide_pb2.Feature(name="", location=request)
+    else:
+      return feature
+```
+
+The method is passed a `route_guide_pb2.Point` request for the RPC and an `RpcContext` object and and returns a `route_guide_pb2.Feature` response.
+
+Now look at the next method. `ListFeatures` is a response-streaming RPC that sends multiple `Feature`s to the client.
+
+```python
+  def ListFeatures(self, request, context):
+    left = min(request.lo.longitude, request.hi.longitude)
+    right = max(request.lo.longitude, request.hi.longitude)
+    top = max(request.lo.latitude, request.hi.latitude)
+    bottom = min(request.lo.latitude, request.hi.latitude)
+    for feature in self.db:
+      if (feature.location.longitude >= left and
+          feature.location.longitude <= right and
+          feature.location.latitude >= bottom and
+          feature.location.latitude <= top):
+        yield feature
+```
+
+Here the request message is a `route_guide_pb2.Rectangle` within which the client wants to find `Feature`s, but instead of returning a single response the method yields zero or more responses.
```

Comment:
Here the request message is a `route_guide_pb2.Rectangle` within which the client wants to find `Feature`s. Instead of returning a single response the method yields zero or more responses.